embed/serve_test.go

```diff
@@ -36,3 +37,33 @@ func TestStartEtcdWrongToken(t *testing.T) {
 		t.Fatalf("expected %v, got %v", auth.ErrInvalidAuthOpts, err)
 	}
 }
+
+// TestStartEtcdLargeBcryptCost ensures that StartEtcd with invalid large bcrypt-cost returns with error.
```

Comment:
Tests in `embed` are no needed. Just unit tests around `NewTokenProvider` should be enough.